frontend/app/mocks/power-platform-api.server.ts

```diff
@@ -244,7 +237,14 @@ export function getPowerPlatformApiMockHandlers() {
         const personSurName = parsedClientApplicationRequest.data.Applicant.PersonName[0].PersonSurName;
         const personBirthDate = parsedClientApplicationRequest.data.Applicant.PersonBirthDate.date;
 
-        const clientApplicationFlags = mockApplicantFlags.get(identificationId);
+        // If the ID is '10000000000', return a 404 error
+        if (identificationId === '10000000000') {
+          log.debug('Client application not found for identification id [%s]', identificationId);
+          return new HttpResponse(null, { status: 404 });
+        }
+
+        // Otherwise, return specific flags or the default
+        const clientApplicationFlags = mockApplicantFlags.get(identificationId) ?? clientApplicationJsonDataSource.BenefitApplication.Applicant.Flags;
 
         if (!clientApplicationFlags) {
```

Comment:
FYI TeamCity build fails:
![image](https://github.com/user-attachments/assets/7e36a802-11d8-4d99-809f-c72680554702)

Reply:
My bad, forgot to remove the old logic.